src/protocol/http/org/apache/jmeter/protocol/http/curl/BasicCurlParser.java

```diff
@@ -252,4 +821,232 @@ public Request parse(String commandLine) {
+    *
+    * @param postdata the post data
+    * @return the result of encoding
+    *
+    */
+   private String encodePostdata(String postdata) {
+       String res = null;
+       if (postdata.contains("@")) {
+           String[] arr = postdata.split("@");
+           try {
+               res = URLEncoder.encode(readFromFile(arr[1]), StandardCharsets.UTF_8.name());
+           } catch (UnsupportedEncodingException e) {
+               LOGGER.error("string '{}' cannot be encoded", readFromFile(arr[1]));// NOSONAR
+           }
+           if (!arr[0].isEmpty()) {
+               res = arr[0] + "=" + res;
+           }
+       } else {
+           if (!postdata.contains("=")) {
+               try {
+                   res = URLEncoder.encode(postdata, StandardCharsets.UTF_8.name());
+               } catch (UnsupportedEncodingException e) {
+                   LOGGER.error("string '{}' cannot be encoded", postdata);
+                   throw new IllegalArgumentException(postdata + " cannot be encoded");
+               }
+           } else {
+               StringBuilder urlAfterEncoding = new StringBuilder();
+               int index = postdata.indexOf('=');
+               urlAfterEncoding.append(postdata.substring(0, index));
+               urlAfterEncoding.append("=");
+               try {
+                   urlAfterEncoding.append(URLEncoder.encode(postdata.substring(index + 1, postdata.length()),
+                           StandardCharsets.UTF_8.name()));
+               } catch (UnsupportedEncodingException e) {
+                   LOGGER.error("string '{}' cannot be encoded", postdata.substring(index + 1, postdata.length()));
+                   throw new IllegalArgumentException(
+                           postdata.substring(index + 1, postdata.length()) + " cannot be encoded");
+               }
+               res = urlAfterEncoding.toString();
+           }
+       }
+       return res;
+   }
+
+   /**
+    * Read the postdata from file
+    *
+    * @param filePath
+    * @return the content of file
+    */
+   private static String readFromFile(String filePath) {
+       String encoding = StandardCharsets.UTF_8.name();
+       String content = "";
+       File file = new File(filePath.trim());
+       if (file.isFile() && file.exists()) {
+           try {
+               content = FileUtils.readFileToString(file, encoding);
+           } catch (IOException e) {
+               LOGGER.error("Failed to read from File {}", filePath);
+               throw new IllegalArgumentException("Failed to read from File " + filePath);
+           }
+       } else {
+           throw new IllegalArgumentException(filePath + " is a directory or does not exist");
+       }
+       return content;
+   }
+
+   /**
+    * Delete line break
+    *
+    * @param postdata the post data
+    * @return the string without break line
+    */
+   private static String deleteLineBreak(String postdata) {
+       String repl = "";
```

Comment:
I'd go for directly returning the results instead of using a variable and reassigning it.